### PR TITLE
bugfix: skip empty lines when calculating indent for auto import

### PR DESCRIPTION
**Pull request by kasiaMarek**
resolves: https://github.com/scalameta/metals/issues/5567

### CODE REVIEW

**Review by jkciesluk — APPROVED** [review]
lgtm

**Review by tgodzik — APPROVED** [review]
LGTM!

File: mtags-shared/src/main/scala/scala/meta/internal/pc/AutoImportPosition.scala [hunk 1]
```diff
@@ -26,11 +28,17 @@ object AutoImportPosition {
...
+  @tailrec
   def inferIndent(lineStart: Int, text: String): Int = {
     var i = 0
     while (lineStart + i < text.length && text.charAt(lineStart + i) == ' ') {
       i += 1
     }
-    i
+
+    val pos = lineStart + i
+    if (pos < text.length() && text.charAt(pos) == '\n') {
```
Comment (filipwiech) [inline mtags-shared/src/main/scala/scala/meta/internal/pc/AutoImportPosition.scala:39]: Just to double check: are we sure that just `\n` is a safe choice that will cover all the cases (for example LF vs CRLF encodings?). Would something like `System.lineSeparator()` be a better choice? :slightly_smiling_face:

Comment (tgodzik) [reply mtags-shared/src/main/scala/scala/meta/internal/pc/AutoImportPosition.scala:39]: Right, always forget about it, maybe we should make sure this works when line ends with `\r\n`

Comment (kasiaMarek) [reply mtags-shared/src/main/scala/scala/meta/internal/pc/AutoImportPosition.scala:39]: I'll fix it and add an lsp test to make sure it works on windows. Thanks for with the reviews @filipwiech, your contribution is much appreciated. 😃

File: mtags-shared/src/main/scala/scala/meta/internal/pc/AutoImportPosition.scala [hunk 2]
```diff
@@ -26,11 +28,18 @@ object AutoImportPosition {
...
+  @tailrec
   def inferIndent(lineStart: Int, text: String): Int = {
     var i = 0
     while (lineStart + i < text.length && text.charAt(lineStart + i) == ' ') {
       i += 1
     }
-    i
+
+    val pos = lineStart + i
+    val accepted = Set('\n', '\r')
```
Comment (filipwiech) [inline mtags-shared/src/main/scala/scala/meta/internal/pc/AutoImportPosition.scala:39]: Thanks for making sure that the support is cross-platform! :+1: Small comment, it seems that this collection will be re-allocated on each invocation of the recursive `inferIndent` function: maybe we could extract it to a static constant and give it a more descriptive name (like `endOfLineCharacters` or some such). :slightly_smiling_face:

File: mtags-shared/src/main/scala/scala/meta/internal/pc/AutoImportPosition.scala [hunk 3]
```diff
@@ -19,18 +21,25 @@ case class AutoImportPosition(
 }
 
 object AutoImportPosition {
+  private val endOfLineCharacters = Set('\t', '\n')
```
Comment (tgodzik) [inline mtags-shared/src/main/scala/scala/meta/internal/pc/AutoImportPosition.scala:24]: `\t` is for tab, it doesn't need to be at the end of the line. Did you mean `\r` ?

Comment (kasiaMarek) [reply mtags-shared/src/main/scala/scala/meta/internal/pc/AutoImportPosition.scala:24]: Yes! I don't know how many times I've tried to get this tiny PR right today 🤯 😮‍💨

**Review by filipwiech — APPROVED** [review]

File: mtags-shared/src/main/scala/scala/meta/internal/pc/AutoImportPosition.scala (hunk 3 above)
Comment (filipwiech) [reply mtags-shared/src/main/scala/scala/meta/internal/pc/AutoImportPosition.scala]: TBH I started to even wonder if perhaps `\t` characters should contribute to the detected indentation (not really sure how and if this is handled elsewhere), but even if that would be the case, then probably it's out of scope of this PR. :sweat_smile:

**Pull request merged**